extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -236,39 +322,67 @@ private function getOrderingMethod() {
 	}
 
 	private function parseAction() {
-		$actions = explode( '/', $this->getPar() );
-		return array_pop( $actions );
+		$pathParts = explode( '/', $this->getPar() );
+		
+		// If there are path parts, try to find the action.  Explode *always* returns an
+		// array even if the content its "exploding" is empty or null.  In those cases it
+		// bizarrely returns an array with one null item.  So just check that the first
+		// element of this array is not empty.
+		if ( !empty( $pathParts[0] ) ) {
+			$action = array_pop( $pathParts );
+			return empty( self::ALLOWED_ACTIONS ) ? self::ACTION_UNREVIEWED : $action;
+		}
+
+		// Return the default action
+		return self::ACTION_UNREVIEWED;
 	}
 
 	private function checkUserPermissions() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)